extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/ProxyConfig.java

```diff
@@ -16,12 +18,22 @@ public class ProxyConfig {
     public boolean proxyAddressForwarding;
 
     /**
-     * If this is true and proxy address forwarding is enabled then the standard {@code Forwarded} header will be used,
-     * rather than the more common but not standard {@code X-Forwarded-For}.
+     * If this is true and proxy address forwarding is enabled then the standard {@code Forwarded} header will be used.
+     * In case the not standard {@code X-Forwarded-For} header is enabled and detected on HTTP requests, the standard header has
+     * the precedence.
      */
     @ConfigItem
     public boolean allowForwarded;
 
+    /**
+     * If either this or {@code allow-forwarded} are true and proxy address forwarding is enabled then the not standard
+     * {@code Forwarded} header will be used.
+     * In case the standard {@code Forwarded} header is enabled and detected on HTTP requests, the standard header has the
```

Comment:
There should be a warning about the security implications of enabling both, i.e. that you need to make sure whatever proxy you have configured knows about both sets of headers, and if it is sending one that is has been configured to strip any other ones from the client.

Reply:
@ejba Yes, it looks fine, my understanding Stuart prefers a bit more specific guidance, for example, you have
```
Activating this together with {@code quarkus.http.proxy.allow-x-forwarded} has security implications as clients can forge
requests with a forwarded header that is not overwritten by the proxy.
```

So may be lets have

```
Activating this together with {@code quarkus.http.proxy.allow-x-forwarded} has security implications as clients can forge
requests with a forwarded header that is not overwritten by the proxy. Therefore proxies should strip unexpected forwarded or x-forwarded headers from the client.
```

I guess something like that ?

Reply:
Agree!